pennylane/interfaces/batch/__init__.py

```diff
@@ -0,0 +1,123 @@
+            for the gradient (if supported).
+        interface (str): The interface that will be used for classical backpropagation.
+            This affects the types of parameters that can exist on the input tapes.
+            Available options include ``autograd``, ``torch``, ``tf``, and ``jax``.
+        accumulation (str): Whether the gradients should be computed on the forward
+            pass (``forward``) or the backward pass (``backward``). Only applies
+            if the device is queried for the gradient; gradient transform
+            functions available in ``qml.gradients`` are only supported on the backward
+            pass.
+
+
+    Returns:
+        list[list[float]]: A nested list of tape results. Each element in
+        the returned list corresponds in order to the provided tapes.
+
+    **Example**
+
+    Consider the following cost function:
+
+    .. code-block:: python
+
+        def cost_fn(params, x, dev):
+            with qml.tape.QuantumTape() as tape1:
+                qml.RX(params[0], wires=0)
+                qml.RY(params[1], wires=0)
+                qml.expval(qml.PauliZ(0))
+
+            with qml.tape.QuantumTape() as tape2:
+                qml.RX(params[2], wires=0)
+                qml.RY(x[0], wires=1)
+                qml.CNOT(wires=[0, 1])
+                qml.probs(wires=0)
+
+            tapes = [tape1, tape2]
+
+            # execute both tapes in a batch on the given device
+            res = execute(tapes, dev)
+
+            return res[0][0] + res[1][0, 0] - res[1][0, 1]
+
+    In this cost function, two **independent** quantum tapes are being
+    constructed; one returning an expectation value, the other probabilities.
+    We then batch execute the two tapes, and reduce the results to obtain
+    a scalar.
+
+    Let's execute this cost function while tracking the gradient:
+
+    >>> dev = qml.device("lightning.qubit", wires=2)
+    >>> params = np.array([0.1, 0.2, 0.3], requires_grad=True)
+    >>> x = np.array([0.5], requires_grad=True)
+    >>> cost_fn(params, x)
+    1.9305068163274222
+
+    Since the ``execute`` function is differentiable, we can
+    also compute the gradient:
+
+    >>> qml.grad(cost_fn)(params, x)
+    (array([-0.0978434 , -0.19767681, -0.29552021]), array([5.37764278e-17]))
+
+    Finally, we can also compute any nth-order derivative. Let's compute the Jacobian
+    of the gradient (that is, the Hessian):
+
+    >>> x.requires_grad = False
+    >>> qml.jacobian(qml.grad(cost_fn))(params, x)
+    array([[-0.97517033,  0.01983384,  0.        ],
+           [ 0.01983384, -0.97517033,  0.        ],
+           [ 0.        ,  0.        , -0.95533649]])
+    """
+    # Default execution function; simply call device.batch execute
+    # and return no Jacobians.
+    execute_fn = lambda tapes: (device.batch_execute(tapes), [])
+
+    if gradient_fn == "device":
+        # gradient function is a device method

```

Comment:
I was not 100% happy with this originally, because it assumes 1 device = 1 gradient method.

However there are ways around this - you could configure the device gradient method at device init, for example:

```pycon
>>> dev = qml.device("some_device", gradient_method="remote_adjoint")
```

This actually has advantages, in that the plugin developer can allow their gradient methods to be configured with full flexibility, rather than having to conform to some restrictive API.

Rather than worry about this too much, I'm happy to leave as-is, and we can revisit when we revisit the device API.